matlabMeas/MeasurementUtilities/initializeSR830Meas2D_Func.m

```diff
@@ -1,73 +1,62 @@
-function [plotHandle] = initializeSR830Meas2D_Func(sweepType1, sweepType1,doBackAndForth)
+function [plotHandle] = initializeSR830Meas2D_Func(sweepTypes, starts, stops, deltas)
 
-[time,Real,Imag] = deal(inf);
+% initializeSR830Meas2D_Func({'Freq', 'ST'}, {1000, 0}, {10000, 1}, {1000, 0.01})
 
-timeLabel = "Time [s]";
-voltageAxisName = genSR830Axis(sweepType);
-yLabel = "Current [A]";
+[sweepType1, sweepType2] = sweepTypes{:};
+[start1, start2] = starts{:};
+[stop1, stop2] = stops{:};
+[deltaParam1, deltaParam2] = deltas{:};
 
-subPlotFigure = figure(getNextMATLABFigNum());
-subplot(2,3,1);
-realVsTime = plotData(time,Real,'xLabel',timeLabel,'yLabel',"Real",'title',"Real vs Time",'subPlot',1);
+xData = [start2, stop2];
+lenX = length(start2:(2 * (start2 < stop2) - 1) * deltaParam2:stop2);
```

Comment:
Interesting way to check the sign of the delta. I have a function that does this, you can change that code such that it uses your logic. It's nice.

Reply:
it'd need to use abs(deltaParam) in order to full work, but i'll replace it where i find it in other code